web-marketplace/src/components/organisms/LocationForm/LocationForm.tsx

```diff
@@ -64,10 +67,20 @@ const LocationForm: React.FC<LocationFormProps> = ({
     name: 'schema:location',
   });
 
+  const [jurisdictionError, setJurisdictionError] = useState<
+    string | undefined
+  >(undefined);
+
   useEffect(() => {
     isDirtyRef.current = isDirty;
   }, [isDirtyRef, isDirty]);
 
+  useEffect(() => {
+    // If a jurisdiction is provided (i.e. the project exists on chain),
+    // we require the location input to match the jurisdiction.
+    setJurisdictionError(validateLocation(jurisdiction, location));
+  }, [jurisdiction, location]);
```

Comment:
In order to get this working I had to check the `location` on render and the `value` on change within the form. I'm not sure if there is a better way of doing this and am a bit confused about how to add custom validation to react hook forms but this should be working as expected now. Let me know if I am overlooking something.

Reply:
If easier done than said, happy to have someone take this over.

Reply:
I think we could have that in the zod schema validation by using `refine` and passing the onChain jurisdiction when instantiating the schema. You can look at the [metadata schema](https://github.com/regen-network/regen-web/blob/dev/web-marketplace/src/components/organisms/MetadataForm/MetadataForm.schema.ts) for an example of how to do that.